libs/utils/dummy_target.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class DummyTarget(object):
+    """
+    Dummy Devlib target
+    """
+
+    def __init__(self, real_target):
+        self._real_target = real_target
+        self._accu_cmds = []
+
+    @property
+    def accumulated_commands(self):
+        return self._accu_cmds
+
+    def clear_accumulator(self):
+        self._accu_cmds = []
+
+    # This is made to look like the devlib Target execute()
+    # Only the command argument is used, the rest is ignored
+    def execute(self, cmd, *rest):
+        self._accu_cmds.append(cmd)
+
+    # Use the DummyTarget methods if they are available
+    # Reach to the real target otherwise
+    def __getattr__(self, name):
+        try:
+            return getattr(super, name)
+        except AttributeError:
+            if self._real_target:
+                return getattr(self._real_target, name)
```

Comment:
I suppose we could, but we'd have to edit that list every time we stumble upon an API that uses a new attribute. I don't see the problem with letting the target run every command but execute - do you have a specific problematic call in mind ? Maybe it would be easier to blacklist them then.